source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,184 @@
+
+.. literalinclude:: /code-snippets/indexes/compound.js
+   :language: js
+   :start-after: begin-ex
+   :end-before: end-ex
+
+Multikey Indexes (Indexes on Array Fields)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Multikey indexes are user-defined ascending or descending indexes on
+fields that contain an array value. You define a multikey index in the
+same fashion as you would a single and compound indexes: you do not 
+need to explicitly specify that an index is multikey.
+
+The following example use the ``createIndex()`` method to create an ascending index on the ``cast`` field in the ``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /code-snippets/indexes/multikey.js
+   :language: js
+   :start-after: begin-ex
+   :end-before: end-ex
+
+Multikey indexes behave differently from non-multikey indexes in terms of
+query coverage, index bound computation, and sort behavior. For a full
+explanation of multikey indexes, including a discussion of their behavior
+and limitations, refer to the :manual:`Multikey Indexes page
+</core/index-multikey>` in the MongoDB manual.
+
+Text Indexes
+~~~~~~~~~~~~
+
+:manual:`Text indexes </core/index-text>` support text search queries on
+string content. ``text`` indexes can include any field whose value is a
+string or an array of string elements. MongoDB supports text search for
+various languages. You can specify the default language as an option when
+creating the index.
+
+The following example uses the ``createIndex()`` method to create a
+``text`` index on the ``fullplot`` field in the ``movies`` collection in
+the ``sample_mflix`` database and specifies the default language to be
+``english``.
+
+.. literalinclude:: /code-snippets/indexes/text.js
+   :language: js
+   :start-after: begin-ex
+   :end-before: end-ex
+
+For a full explanation of text search with MongoDB, refer to :manual:`Text Indexes </core/index-text>` in the MongoDB manual.
+
+Geospatial Indexes
+~~~~~~~~~~~~~~~~~~
+
+MongoDB supports queries of geospatial coordinate data using :manual:`2dsphere </core/2dsphere>` indexes. With a ``2dsphere`` index, you can query for inclusion, intersection, and proximity. The :manual:`Geospatial Queries page </geospatial-queries>` in the MongoDB manual provides a detailed explanation of geospatial query operations in MongoDB.
+
+To create a ``2dsphere`` index, the field you are indexing must be a :manual:`GeoJSON object </reference/geojson>`, as in the following sample document from the ``theaters`` collection in the ``sample_mflix`` database:
+
+.. code-block:: javascript
+
+   {
+      "_id" : ObjectId("59a47286cfa9a3a73e51e75c"),
+      "theaterId" : 104,
+      "location" : {
+         "address" : {
+            "street1" : "5000 W 147th St",
+            "city" : "Hawthorne",
+            "state" : "CA",
+            "zipcode" : "90250"
+         },
+         "geo" : {
```

Comment:
For an example, I feel like we might be better off ditching all the fields besides "geo" or at least simplifying them. Right now I feel like this makes it seem like geoJSON objects have a lot more data than they really do -- maybe we can meet in the middle and more specifically call out the location.geo field before throwing users into this example?